### ISSUE TITLE
Speed-up of quantitative genetics models

### ISSUE THREAD

**Issue by mikheyev** [paul-buerkner/brms#541]
In the field of quantitative genetics it is common to have a large matrix as a random covariate. These models [run quite slow in brms](https://github.com/JonBrommer/Multivariate-Mixed-Models-in-R/wiki/brms-examples#organisational-level-2), compared to [comparable models in MCMCglmm](https://github.com/mikheyev/brms-wam). If the performance could be increased (fairly dramatically), because of its flexibility and easier syntax brms would be extremely useful for quantitative genetics analysis. I am not sure whether this is an issue with Stan or with brms, or how easy it would be to actually implement... but it would be useful.

**Comment by paul-buerkner** [comment]
Can you be more specific on what you mean by "quite slow" as compared to MCMCglmm? Which of these models or all?

**Comment by mikheyev** [comment]
You can see an example [here](https://mikheyev.github.io/brms-wam/), based on your phylogenetic example, but with a much bigger matrix.

**Comment by paul-buerkner** [comment]
You are right. brms/Stan is rather slow as compared to MCMCglmm for this example. The model, in general, is rather hard to fit because each observation has two residuals (one correlated and one uncorrelated) battling for the variance of the response. This is what causes brms/Stan to be slow.

MCMCglmm has an algorithm that is tuned to such models (i.e., uses block Gibbs sampling of varying effects if I remember correctly), which -- if it works -- is hard to beat. However, for more complex models (with out without phylogenetic structure), you will have huge problems estimating the model with MCMCglmm at all. 

Long story short, I don't think there is much to improve from the brms side here, but you can open a thread on https://discourse.mc-stan.org/ and maybe someone has an idea.

**Comment by mikheyev** [comment]
OK, will do. As I'm deeply unfamiliar with Stan, any do you have any advice on how to formulate the question? Sorry for the ignorance...

**Comment by paul-buerkner** [comment]
You could either post the Stan code generated by brms on discourse and ask for suggestions on how to improve the phylogenetic part or (perhaps better) could ask people directly how they implement this kind of model in Stan.

**Comment by paul-buerkner** [comment]
I am closing this issue for now. Feel free to re-open it if something comes up that we hope can improve the efficiency of the discussed models.
